library/kotlin/src/io/envoyproxy/envoymobile/UpstreamHttpProtocol.kt

```diff
@@ -5,5 +5,15 @@ package io.envoyproxy.envoymobile
  */
 enum class UpstreamHttpProtocol(internal val stringValue: String) {
   HTTP1("http1"),
-  HTTP2("http2"),
+  HTTP2("http2");
+
+  companion object {
+    fun enumValue(stringRepresentation: String): UpstreamHttpProtocol {
+      return when (stringRepresentation) {
+        "http1" -> UpstreamHttpProtocol.HTTP1
+        "http2" -> UpstreamHttpProtocol.HTTP2
+        else -> throw IllegalArgumentException("Unable to find value for $stringRepresentation")
```

Comment:
I noticed that some of these throw and others are nullable. Should we keep them consistent?

Reply:
We had it that way because of the automatic addition of "retriable-status-codes", although I'd rather throw in all of them. I have updated to something I think is better.